NPE while loading Healthcheck controller

```
SEVERE: Unable to wait until the task is completed
java.util.concurrent.ExecutionException: org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.TextField.textProperty()" because "this.searchText" is null
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
	at org.eclipse.fx.ui.controls.internal.FXThreadSynchronizeImpl.syncExec(FXThreadSynchronizeImpl.java:114)
	at org.eclipse.fx.ui.workbench.fx.internal.UISynchronizeImpl.syncExec(UISynchronizeImpl.java:51)
	at org.eclipse.e4.ui.internal.di.UIEventObjectSupplier$UIEventHandler.handleEvent(UIEventObjectSupplier.java:64)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:205)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:203)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:345)
Caused by: org.eclipse.e4.core.di.InjectionException: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.TextField.textProperty()" because "this.searchText" is null
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:68)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:457)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:456)
	at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
Caused by: java.lang.NullPointerException: Cannot invoke "javafx.scene.control.TextField.textProperty()" because "this.searchText" is null
	at com.osgifx.console.ui.healthchecks.HealthCheckFxController.initSearchFilter(HealthCheckFxController.java:197)
	at com.osgifx.console.ui.healthchecks.HealthCheckFxController.initNames(HealthCheckFxController.java:155)
	at com.osgifx.console.ui.healthchecks.HealthCheckFxController.onUnderlyingDataUpdate(HealthCheckFxController.java:319)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:58)
	... 6 more
```

--- a/com.osgifx.console.ui.healthchecks/src/main/java/com/osgifx/console/ui/healthchecks/HealthCheckFxController.java
+++ b/com.osgifx.console.ui.healthchecks/src/main/java/com/osgifx/console/ui/healthchecks/HealthCheckFxController.java
@@ -33,9 +33,10 @@ import org.apache.commons.lang.StringUtils;
 import org.controlsfx.control.CheckListView;
 import org.controlsfx.control.MaskerPane;
 import org.controlsfx.control.SegmentedButton;
-import org.eclipse.e4.core.di.annotations.Optional;
-import org.eclipse.e4.ui.di.UIEventTopic;
+import org.eclipse.fx.core.Subscription;
 import org.eclipse.fx.core.ThreadSynchronize;
+import org.eclipse.fx.core.event.EventBus;
+import org.eclipse.fx.core.event.Topic;
 import org.eclipse.fx.core.log.FluentLogger;
 import org.eclipse.fx.core.log.Log;
 import org.osgi.annotation.bundle.Requirement;
@@ -97,9 +98,12 @@ public final class HealthCheckFxController {
 	private Supervisor            supervisor;
 	@Inject
 	private ThreadSynchronize     threadSync;
+	@Inject
+	private EventBus              eventBus;
 	private MaskerPane            progressPane;
 	private ExecutorService       executor;
 	private Future<?>             hcExecFuture;
+	private Subscription          subscription;
 
 	@FXML
 	public void initialize() {
@@ -110,9 +114,27 @@ public final class HealthCheckFxController {
 			logger.atDebug().log("FXML controller has been initialized");
 			initHcTypeButton();
 			initButtons();
+			listenToDateRetrieval();
 		} catch (final Exception e) {
 			logger.atError().withException(e).log("FXML controller could not be initialized");
 		}
+	}
+
+	/**
+	 * This e(fx)clipse way to listening to events is done as the usual e4 way using
+	 * event annotations doesn't work since the annotated event method is called at
+	 * at the very first time before the FXML elements are even injected and
+	 * therefore, throws NPE.
+	 */
+	private void listenToDateRetrieval() {
+		subscription = eventBus.subscribe(new Topic<>(DATA_RETRIEVED_HEALTHCHECKS_TOPIC), event -> {
+			if (nameHcButton.isSelected()) {
+				initNames();
+			} else {
+				tagHcButton.setSelected(true);
+				initTags();
+			}
+		});
 	}
 
 	private void initHcTypeButton() {
@@ -163,6 +185,7 @@ public final class HealthCheckFxController {
 	@PreDestroy
 	public void destroy() {
 		executor.shutdownNow();
+		subscription.dispose();
 	}
 
 	private void initHcList() {
@@ -289,17 +312,6 @@ public final class HealthCheckFxController {
 	@FXML
 	private void deselectAll(final ActionEvent event) {
 		hcMetadataList.getCheckModel().clearChecks();
-	}
-
-	@Inject
-	@Optional
-	private void onUnderlyingDataUpdate(@UIEventTopic(DATA_RETRIEVED_HEALTHCHECKS_TOPIC) final String data) {
-		if (nameHcButton.isSelected()) {
-			initNames();
-		} else {
-			tagHcButton.setSelected(true);
-			initTags();
-		}
 	}
 
 }

